feat(MTMR): document the new MTMR conversion tool (#2080)
bonitasoft/bonita-doc: modules/version-update/pages/mtmr-tool.adoc

= Bonita Multi-Tenant to Multi-Runtime conversion tool
:description: This pages describes how to use the multi-tenant to multi-runtime conversion tool

{description}

Learn how to use the multi-tenant to multi-runtime tool to convert a multi-tenant platform into a multi-runtime one.

WARNING: xref:ROOT:multi-tenancy-and-tenant-configuration.adoc[Multi-tenancy] is deprecated starting from Bonita 2021.1 (technical version 7.12) and will be removed technically with Bonita 2023.1 (technical version 7.16).
*If you are using multi-tenancy, you will have to run this conversion tool before updating to 7.16.*

image:version-update:mtmr-general-idea-schema.svg[General idea of the conversion tool,width=700]

== Overview

Following the commercial depreciation of multi-tenancy use by customer willing to have separate BDMs, the technical depreciation will be effective with Bonita 2023.1. The technical solution for customers willing to have separate BDMs per applications/processes proposed by Bonitasoft will be multi-runtimes. 

> In order to prepare the technical conversion from multi-tenancy to multi-runtime, Bonitasoft developped a dedicated tool, that will have to be run as many times as the number of the tenants to be kept. 

Author: ```suggestion
In order to prepare the technical conversion from multi-tenancy to multi-runtime, Bonitasoft developped a dedicated tool, that will have to be run once for every tenant to be kept. 
```

> The tool will delete everything not related to the tenant specified in the parameter from the given database, this means you will need to dump your current database and mount one new database dump per tenant in your environment. 

Reviewer: Some confusions for me:
- To prevent some misunderstanding, maybe we should explicitly say that the customer will need to set up a new database instance per tenant. And then load the dump onto it.
- By "environment" I might understand a runtime server and a database server. But we can use the same runtime server to run multiple runtime instances.

Reviewer 2: Suggestion accepted :)

NOTE: The tool will have to be executed for every database dump.

== How works Bonita Multi-tenant to Multi-Runtime conversion tool 
=== Pre-requisites
* The tool requires Java 11+ JRE
* A database of a Bonita Multi-tenant Platform. +

> The aim is to restore a dump of your production database where you want one of your new Bonita Runtimes to run in the future, and then to run the tool on it.

Reviewer: Suggestion: move this sentence in the overview and add a new prerequisites line saying something like "A dump, from actual production database, loaded to the new database"

Reviewer 2: Suggestion accepted :)

Learn how to use the multi-tenant to multi-runtime tool to convert a multi-tenant platform into a multi-runtime one.

WARNING: xref:ROOT:multi-tenancy-and-tenant-configuration.adoc[Multi-tenancy] is deprecated starting from Bonita 2021.1 (technical version 7.12) and will be removed technically with Bonita 2023.1 (technical version 7.16).
*If you are using multi-tenancy, you will have to run this conversion tool before updating to 7.16.*

image:version-update:mtmr-general-idea-schema.svg[General idea of the conversion tool,width=700]

== Overview

> Following the commercial depreciation of multi-tenancy use by customer willing to have separate BDMs, the technical depreciation will be effective with Bonita 2023.1. The technical solution for customers willing to have separate BDMs per applications/processes proposed by Bonitasoft will be multi-runtimes. 

Author: "for customers willing to have separate BDMs per applications/processes" --> I am not certain that is correct.
If it means BDM as in the BDM database, it's just wrong/weird.
If it means something else in this context, we need to use another term or explicit the abbreviation, because BDM in the documentation always means the BDM database.

Reviewer 2: Changed it with Bonita engine database. Adrien's suggestion :)
